blocks/media-upload-button/index.js

```diff
@@ -77,13 +34,9 @@ class MediaUploadButton extends Component {
 			frameConfig.library = { type };
 		}
 
-		if ( gallery ) {
-			const GalleryDetailsMediaFrame = getGalleryDetailsMediaFrame();
-			this.frame = new GalleryDetailsMediaFrame( {
-				frame: 'select',
-				mimeType: type,
-				state: 'gallery',
-			} );
+		// Use a frame workflow passed to the button if it exists.
+		if ( frame ) {
+			this.frame = new frame();
```

Comment:
See my previous comment about the redundancy of other props when passing a frame directly. Again, I don't think this is ideal from a dev UX.